plutus-errors/exe-bootstrap/TH/Bootstrap.hs

```diff
@@ -0,0 +1,75 @@
+-- | (Re)generate ErrorCode instances for initial errors
+{-# LANGUAGE TemplateHaskell #-}
+module TH.Bootstrap (
+    bootstrap
+    ) where
+
+import           Data.Foldable
+import           Data.Map                     as M
+import           ErrorCode
+import           Language.Haskell.TH
+import           Language.Haskell.TH.Datatype
+import           Numeric.Natural
+
+{- |
+The purpose of this function is to help in the (re)-generation of 'ErrorCode' instances
+for Plutus-errors/data-constructors. The user can call this function as a script to
+get the generated instances as Haskell code and paste/modify it accordingly next to the errors (for avoiding orphans).
+The function works by assigning a unique number to each dataconstructor, starting counting from 1.
+The function groups the data-constructors by their "parent" type-constructor,
+so the order that they are given in the input list does not matter.
+-}
+bootstrap :: [Name] -> Q Exp
+bootstrap constrs = do
+    -- give them a unique numbering
+    let indexedCs = zip constrs $ [1::Natural ..]
+    -- group them by their parent type
+    groups <- groupConstrs indexedCs
+    -- generate the instances and Haskell-prettyprint them
+    let decs = StringL $ pprint $ M.elems $ M.mapWithKey makeInstance groups
+    [| putStr $(litE decs) |]
+
+-- | A dataconstructor representing a plutus error, paired with a generated unique errorcode
+type IxCons = (Name, Natural)
+-- | An intermediate structure to group data-constructors by their parent type-constructor and number of tyvars
+type Group = (ParentName,Int)
+
+-- | Groups a list of (possibly-unrelated) dataconstructors by their parent typeconstructor
+groupConstrs :: [IxCons] -> Q (Map Group [IxCons])
```

Comment:
This seems a bit convoluted to me: we give all the constructors numbers, then we group them (with their already-assigned numbers), then we do more stuff...

In fact, I am pretty sure the comment is false ("The function groups the data-constructors by their "parent" type-constructor, so the order that they are given in the input list does not matter."), and in fact if the argument contained constructors from different types spread through the list, they'd get a weird numbering.

What if we:
1. Grouped the constructors by parent dataconstructor.
2. Flattened that list and *then* numbered them.

I guess it doesn't matter too much, since we won't reuse this function much...